src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -402,10 +402,15 @@ public override Symbol VisitAssembly(AssemblySymbol symbol)
                     return _otherAssembly;
                 }
 
-                // find a referenced assembly with the exactly same identity:
+                // find a referenced assembly with the exactly same source identity:
                 foreach (var otherReferencedAssembly in _otherAssembly.Modules[0].ReferencedAssemblySymbols)
                 {
-                    if (symbol.Identity.Equals(otherReferencedAssembly.Identity))
+                    var identity = symbol.Identity;
+                    var otherIdentity = otherReferencedAssembly.Identity;
+
+                    if (AssemblyIdentityComparer.SimpleNameComparer.Equals(identity.Name, otherIdentity.Name) &&
+                        symbol.AssemblyVersionInSource.Equals(otherReferencedAssembly.AssemblyVersionInSource) &&
```

Comment:
Not here. If we are not able to retrieve the version in source and the versions don't match we report an error earlier in the pipeline (where we construct the version mapping). Also note that EnC always requires the project to exist in the solution, so we could only run into these issues if someone built the project with a previous compiler while having it open in VS and VS wouldn't rebuild it before starting the debugging session. If you were malicious you could probably achieve that but it would take an extra effort.